components/client/typescript/src/index.ts

```diff
@@ -20,29 +102,41 @@ import {
  */
 export class ChainhookEventObserver {
   private fastify?: FastifyInstance;
-  private serverOpts: ServerOptions;
-  private chainhookOpts: ChainhookNodeOptions;
+  private observer: EventObserverOptions;
+  private chainhook: ChainhookNodeOptions;
+  private healthCheckTimer?: NodeJS.Timer;
 
-  constructor(serverOpts: ServerOptions, chainhookOpts: ChainhookNodeOptions) {
-    this.serverOpts = serverOpts;
-    this.chainhookOpts = chainhookOpts;
+  constructor(observer: EventObserverOptions, chainhook: ChainhookNodeOptions) {
+    this.observer = observer;
+    this.chainhook = chainhook;
   }
 
   /**
-   * Start the Chainhook event server.
-   * @param predicates - Predicates to register
+   * Starts the Chainhook event observer.
+   * @param predicates - Predicates to register. If `predicates_disk_file_path` is enabled in the
+   * observer, predicates stored on disk will take precedent over those specified here.
    * @param callback - Function to handle every Chainhook event payload sent by the node
    */
-  async start(predicates: ServerPredicate[], callback: OnEventCallback): Promise<void> {
+  async start(
+    predicates: EventObserverPredicate[],
+    callback: OnPredicatePayloadCallback
+  ): Promise<void> {
     if (this.fastify) return;
-    this.fastify = await buildServer(this.serverOpts, this.chainhookOpts, predicates, callback);
-    await this.fastify.listen({ host: this.serverOpts.hostname, port: this.serverOpts.port });
+    this.fastify = await buildServer(this.observer, this.chainhook, predicates, callback);
+    await this.fastify.listen({ host: this.observer.hostname, port: this.observer.port });
+    if (this.observer.predicate_health_check_interval_ms && this.healthCheckTimer === undefined) {
+      this.healthCheckTimer = setInterval(() => {
+        void predicateHealthCheck(this.observer, this.chainhook);
```

Comment:
Can `predicateHealthCheck` ever throw? If so, this needs a `.catch(err => log(err))` otherwise could end up with an unhandled promise rejection